drafts/builder-pattern-tables.md

```diff
@@ -0,0 +1,354 @@
+  | TableCellContentValueList
+  | TableCellContentIcon
+  | TableCellContentButton
+  | undefined;
+
+export type TableColumnConfiguration<T> = {
+  columnId: string;
+  title: string;
+  cellContent: {
+    displayType: TableCellDisplayingType;
+    valueGetterFn: (data: T) => TableCellContentTypes;
+    formatFn: (cellValue: TableCellContentTypes) => TableCellContentTypes;
+  };
+  styling: {
+    width?: string;
+    minWidth?: string;
+    maxWidth?: string;
+  };
+};
+```
+
+The `cellContent` attribute defines a value getter function specifying how to retrieve the required cell value from our data object. We can also add formatting functions here to e.g. transform the fetched value to a date or currency.
+
+The `displayType` attribute is used to switch between icons, buttons, string lists (multi-line), or simple string values displaying inside the table cell. Of course, this could easily be extended with other types that you want to display in your table cells, like references or even nested tables (please don't do that). 
+
+The `styling` attribute helps us to improve the responsiveness of the columns by defining min and max width (if required), or even fixed-width columns.
+
+So, when we look at the "Created" date column of our example table, this would yield a column configuration as follows:
+
+```json
+{
+  "title": "Created",
+  "columnId": "created-date",
+  "cellContent": {
+    "displayType": TableCellContentTypes.TableCellContentValue,
+    "valueGetterFn": (date: ExampleTO) => date.created,
+    "formattingFn": (date?: string) => formatDate(date, 'mediumShort')
+  },
+  "styling": {
+    "minWidth": "105px",
+    "maxWidth": "250px"
+  }
+}
+```
+
+# Builder Pattern
+
+There are not a lot of use cases where I would consider using a builder pattern in front-end applications. However, when dealing with the column configurations of our shared table component, it comes in quite handy. The builder pattern allows us to easily create such column definitions in a readable and scalable fashion.
+
+```typescript
+public initTable{
+    const builder = new TableColumnConfigBuilder<ExampleTO>();
+    this.columnDefinitions = [
+        builder.column('ID').id((data) => data.individualId.id), 
+        builder.column('Products').multiLineList((data) => data.products),
+        builder.column('Action').button('Open', onButtonClickFn),
+        builder.column('Created').date((data) => data.created_at)
+    ]
+}
+```
+
+So, every time you add a table to your component, you need to specify all the columns of your table using the `TableColumnConfigBuilder`. The builder itself is fairly straightforward. For each build step, you simply modify the generic column definition by adding, removing, or changing attributes. For example, in the build step `date()` we define a value getter, a min. and max. width and change the formatting of the cell to date formatting. Below is an excerpt of our table builder with a few sample steps:
```

Comment:
```suggestion
So, every time you add a table to your component, you need to specify all the columns of your table using the `TableColumnConfigBuilder`. The builder itself is fairly straightforward. For each build step, you simply modify the generic column definition by adding, removing, or changing attributes. For example, in the build step `date()` we define a value getter, a `min` and `max` width and change the formatting of the cell to date formatting. Below is an excerpt of our table builder with a few sample steps:
```
**suggestion:** maybe like this? the dots are a bit confusing as I wouldn't see them as abbreviations (at least not in this context where we're used to the terms "min" and "max")